delayrecorder.py

```diff
@@ -1,87 +1,175 @@
+    if quietMode:
+        print("Running in quiet mode, won't print the measurements to the terminal")
+
+    # Read messages from Arduino
+    timeout = time.time() + 0.01
+    while True:
+        a = ser.readline().decode()
+        if time.time() > timeout:
+            break
+
+    measurements = []
+    i = 0
+    overallrounds = 0
+    initmessage = 0
+
+    while (i < num_measurements):
+        overallrounds = overallrounds + 1
+        a = ser.readline().decode()
+        if "." in a:
+            initmessage = 1
+            i = i + 1
+            a = a.replace("\n","")
+            a = a.replace("\r","")
+            measurements.append(float(a))
+            if not quietMode:
+                print("G2G Delay trial " + str(i) + "/" + str(num_measurements) + ": " + a + " ms")
+            else:
+                print("G2G Delay trial " + str(i) + "/" + str(num_measurements), end='\r')
+        else:
+            if overallrounds > 0 and initmessage == 1:
+                print("Did not receive msmt data from the Arduino for another 5 seconds. Is the phototransistor still sensing the LED?")
+            else:
+                print("Did not receive msmt data from the Arduino for 5 seconds. \n Is the phototransistor sensing the LED on the screen? \n Is the correct side of the PT pointing towards the screen (the flat side with the knob on it)? \n Is the screen brightness high enough (max recommended)?")
+                initmessage = 1
+
+    return measurements
+
+def write_measurements_csv(filename, measurements, stats):
+    with open(filename, 'w') as f:
+        writer = csv.writer(f)
+        writer.writerow(measurements)
+        header = ['Samples', 'Min', 'Max', 'Mean', 'Median', 'Std']
+        writer.writerow(header)
+        stats_list = [stats.num_measurements, stats.min, stats.max, stats.mean, stats.median, stats.std]
+        writer.writerow(stats_list)
+    print("Saved results to " + args.filename)
+
+def get_measurements_csv(filename):
+    with open(filename, 'r') as f:
+        reader = csv.reader(f)
+        # first row has the datapoints, 3rd row has the stats
+        for i, row in enumerate(reader):
+            if i == 0:
+                measurements = [float(i) for i in row]
+            if i == 2:
+                # convert every string in the array into a float
+                stats_list = [float(i) for i in row]
+                # grab each element of the list to create the Stats object
+                stats = Stats(*stats_list)
+    print("Obtained values from " + filename)
+
+    return measurements, stats
+
+def generate_stats(measurements):
+    # Obtain general stats
+    measurements_np = np.array(measurements)
+    min = np.min(measurements_np)
+    max = np.max(measurements_np)
+    mean = np.mean(measurements_np)
+    median = np.median(measurements_np)
+    std = np.std(measurements_np)
+    print("Min/Max G2G Delay: " + '%.2f' % (min) + "/" + '%.2f' % (max) + " ms")
```

Comment:
Let's use f-strings! :)

```suggestion
    print(f"G2G Delay: min={min} ms, max={max} ms")
```

Reply:
Also consider introducing f-strings throughout the file, where useful